pyccel/codegen/printing/pycode.py

```diff
@@ -846,28 +846,37 @@ def _print_NumpyCountNonZero(self, expr):
         return "{}({})".format(name, arg)
 
     def _print_ListAppend(self, expr):
-        method_name = expr.name
+        method_name = self._print(expr.name)
         list_var = self._print(expr.list_variable)
         append_arg = self._print(expr.append_argument)
 
         return f"{list_var}.{method_name}({append_arg})\n"
 
+    def _print_ListExtend(self, expr):
+        method_name = self._print(expr.name)
+        list_var = self._print(expr.list_variable)
+        extend_arg = self._print(expr.extend_argument)
+
+        return f"{list_var}.{method_name}({extend_arg})\n"
+
     def _print_ListInsert(self, expr):
-        method_name = expr.name
+        method_name = self._print(expr.name)
         index = self._print(expr.index)
         list_var = self._print(expr.list_variable)
         insert_arg = self._print(expr.insert_argument)
 
         return f"{list_var}.{method_name}({index}, {insert_arg})\n"
 
     def _print_ListPop(self, expr):
+        method_name = self._print(expr.name)
         args = self._print(expr.pop_index) if expr.pop_index else ""
-        name = self._print(expr.list_variable)
-        return f"{name}.pop({args})"
+        list_var = self._print(expr.list_variable)
+        return f"{list_var}.{method_name}({args})"
 
     def _print_ListClear(self, expr):
+        method_name = self._print(expr.name)
         name = self._print(expr.list_variable)
-        return f"{name}.clear()\n"
+        return f"{name}.{method_name}()\n"
```

Comment:
This is a lot of similar methods. Maybe it would make sense to create a superclass :
```
class ListMethod(PyccelInternalFunction):
    __slots__ = ('_list_variable',)
    def __init__(self,  list_variable, *args):
        self._list_variable = list_variable
        super().__init__(*args)

    @property
    def list_variable(self):
        """
        Get the variable representing the list.
        Get the variable representing the list.
        """
        return self._list_variable
```
To avoid this duplication?

Reply:
I agree, i'll update the classes

Reply:
implemented the superclass and it's printer in 18e88d2 and fa3da48